dashboard/app/controllers/registrations_controller.rb

```diff
@@ -9,32 +9,69 @@ def new
 
   def update
     return head(:bad_request) if params[:user].nil?
-    @user = User.find(current_user.id)
+    current_user.reload # Needed to make tests pass for reasons noted in registrations_controller_test.rb
 
     successfully_updated =
-      if forbidden_change?(@user, params)
+      if forbidden_change?(current_user, params)
         false
-      elsif needs_password?(@user, params)
-        @user.update_with_password(update_params(params))
+      elsif needs_password?(current_user, params)
+        current_user.update_with_password(update_params(params))
       else
         # remove the virtual current_password attribute update_without_password
         # doesn't know how to ignore it
         params[:user].delete(:current_password)
-        @user.update_without_password(update_params(params))
+        current_user.update_without_password(update_params(params))
       end
 
+    respond_to_account_update(successfully_updated)
+  end
+
+  def create
+    Retryable.retryable on: [Mysql2::Error, ActiveRecord::RecordNotUnique], matching: /Duplicate entry/ do
+      super
+    end
+  end
+
+  def upgrade
+    return head(:bad_request) if params[:user].nil?
+    user_params = params[:user]
+    user_params[:provider] = nil
```

Comment:
Also, this could be one-lined to `user_params = params[:user].merge(provider: nil)`.